.github/workflows/version-bump.yml

```diff
@@ -0,0 +1,93 @@
+      # This is to support testing release automation in forks, it is disabled for the canonical docs repo
+      - name: Override ${{github.event.inputs.project}}'s submodule URL
+        if: vars.IS_CANONICAL_DOCS_REPO != 'true' && github.event.inputs.project-fork-url != ''
+        run: git config --local --add "submodule.src/$PROJECT.url" "$FORK_URL"
+        env:
+          FORK_URL: ${{github.event.inputs.project-fork-url}}
+
+      # ----------------------------------------------------------------------- Update the submodule
+      - name: Clone ${{github.event.inputs.project}} submodule
+        run: git submodule update --init "src/$PROJECT/"
+      - name: Update ${{github.event.inputs.project}} submodule
+        working-directory: src/${{github.event.inputs.project}}/
+        run: git checkout "refs/tags/$VERSION"
+      - name: Stage ${{github.event.inputs.project}} submodule
+        run: git add "src/$PROJECT/"
+
+      # ----------------------------------------------------------------------- Update the installation page
+      - name: Setup Python 3.10
+        if: github.event.inputs.project == 'bonsai'
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Update Bonsai version info
+        if: github.event.inputs.project == 'bonsai'
+        run: |
+          python .github/workflows/update-bonsai-version-info.py
+          git add articles/installation.md
+        env:
+          workflow_dispatch_version: ${{github.event.inputs.version}}
+          is_canonical_docs_repo: ${{vars.IS_CANONICAL_DOCS_REPO}}
+          project_fork_url: ${{github.event.inputs.project-fork-url}}
+
+      # ----------------------------------------------------------------------- Commit changes
+      # Skip the rest of the job if there aren't any changes to commit
+      # (IE: the submodule was already the relevant version)
+      - name: Check if update was necessary
+        id: pre-commit-check
+        run: |
+          (git diff-index --cached --exit-code HEAD \
+            && python .github/workflows/gha.py print_notice "Version bump was no-op, no changes to commit.") \
+            || python .github/workflows/gha.py set_output continue true
+      - name: Commit changes
+        if: steps.pre-commit-check.outputs.continue == 'true'
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git commit -m "Update \`$PROJECT\` to \`$VERSION\`"
+      - name: Push changes
+        if: steps.pre-commit-check.outputs.continue == 'true'
+        run: git push
+      # The above push will not actually trigger a deployment as actions performed using temporary GitHub Actions tokens
+      # do not trigger events in order to avoid unintentional recursion. As such we manually trigger deployment.
+      - name: Trigger GitHub Pages deployment
```

Comment:
An unfortunate quirk of how this step works is that the deployment workflow run will be attributed to the GitHub Actions bot. This means if it fails for some reason then nobody will be notified and the failure might go unnoticed.

We could provide a personal access token instead, but I don't love using them unless absolutely necessary. This would attribute all deployments to whoever made the PAT, which means they'd get notifications related to it.

Another option is to send a notification outside of GitHub when the deployment fails. This is what [I've done in the past using Microsoft Teams](https://github.com/MochiLibraries/Biohazrd/blob/4dbaabfb8195853a931a21d9b6468baed9020bf0/.github/workflows/send-ci-failure-notification.py) back when Actions was still in beta and didn't have failure notifications yet.

The simplest option (other than doing nothing) might be to display the status badge somewhere and hope someone notices it. Would need to double-check that it updates for manually-run workflows though, not sure that it does. (IE: ![Build documentation](https://github.com/bonsai-rx/docs/actions/workflows/build.yml/badge.svg))

Reply:
The badge seems like a nice and standard solution to this, for now I should notice it either way (now that I think about it, it probably would be nice to have a badge for bonsai release automation too on the main README.md).

Reply:
I will add a badge to both!

Reply:
[![](https://img.shields.io/github/actions/workflow/status/bonsai-rx/docs/build.yml?branch=main&style=flat-square&label=Deployment%20status)](https://github.com/bonsai-rx/docs/actions/workflows/build.yml)